test/test_sparse.py

```diff
@@ -3566,12 +3566,13 @@ def test(sparse_dims, nnz, with_size, new_size):
     @dtypes(*all_types_and_complex_and(torch.bool))
     def test_sparse_spdiags(self, device, dtype):
 
-        make_diags = functools.partial(make_tensor, dtype=dtype, device=device)
+        requires_grad = dtype in (torch.double, torch.cdouble)
+        make_diags = functools.partial(make_tensor, dtype=dtype, device=device, requires_grad=requires_grad)
         make_offsets = functools.partial(torch.tensor, dtype=torch.long, device=device)
 
         if TEST_SCIPY:
             def reference(diags, offsets, shape):
-                return scipy.sparse.spdiags(diags, offsets, *shape).toarray()
+                return scipy.sparse.spdiags(diags.detach().cpu(), offsets.cpu(), *shape).toarray()
```

Comment:
Just curious, is `detach` necessary here?

Reply:
When passing the tensor to SciPy there will be a `.numpy()` conversion done. If `diags` has `requires_grad=True` this causes a runtime error: `Can't call numpy() on a Tensor that requires grad. Use tensor.detach().numpy() instead`.